api/bundles/org.jboss.tools.rsp.api/src/main/java/org/jboss/tools/rsp/api/dao/DeployableReference.java

```diff
@@ -42,9 +42,6 @@ public String getPath() {
 		return path;
 	}
 
-	public void setPath(String path) {
```

Comment:
dao's must be java beans, which must have all getters and setters so the framework can instantiate it properly. Please re-add this method.

Reply:
ok, right, but then, how to prohibit a change in path for the reference?
Given that the deployable reference is stored in a map by path, I dont think that allowing to change the path once it's in the map is a good idea?

Reply:
btw. interestingly DaoClassesTest isnt failing even though we're missing the setter.

Reply:
There's only 4 possible entities calling setPath. 

1) The client. The client may use setPath for 2 reasons. 
     a) To instantiate it for sending to the server. This is a valid use of setPath
    b) Setting the path on an object they received FROM the server. This is a nonsensical use of the API and has no effect to anything on the server. There is no reason to do this. 
2) lsp4j, either in seriailziing or unserializing. This is a valid (and required) use of the API. 
3) rsp-server itself.  The rsp-server framework should never really be calling setPath on a path it receives from the client, but there's no real ability to prevent them from doing it. Either way, it should be assuemd the framework itself is allowed to do whatever it wants. 
4) An extender to rsp-server. An extender to rsp-server such as wildfly could technically call setPath on a path received from the client before storing it in its own models. There is no valid reason to do this. However, even if we tried to restrict this usecase by not making setPath public, the extender could instead simply instantiate a new object with identical fields and a different path. Therefore, restricting access to setPath would achieve no benefit. 

We do store some daos in the models, specifically in the ServerPublishStateModel. The best way to protect against people changing the variables stored in the model are to only return clones of those objects wherever possible. While getDeployableState(reference) currently does return a clone,  getDeployableStates() returns a list with the same objects. It might be better to clone not just the list but each item, here. 

In general, java beans need to be protected from malicious usage. Usually this is accomplished by not storing them in the models, but this then requires costly conversion back and forth, from the dao to a storable class, and then back. The second best thing is to not return any object that is directly stored in a model by cloning everything when getters are accessed. 

Either way, we cannot remove setPath from the dao, as it is a java bean / dao, and is required. 
The only entities capable of changing it once it is stored in this map are either rsp-server itself, or the custom server adapter extension. In both cases, I would argue that both are allowed to do so, but probably have no valid use case to do so and thus should not do it. But this still doesn't mean we can remove or block the API.

Reply:
moved this item to issue #310